snuba/datasets/configuration/transactions/dataset.yaml

```diff
@@ -0,0 +1,8 @@
+version: v1
+kind: dataset
+name: transactions
+
+is_experimental: false
```

Comment:
Is this field mandatory for all datasets? Can we make it optional such that only new datasets need to set it up as experimental.

Reply:
The field itself is not required by json_schema: https://github.com/getsentry/snuba/blob/3d12cf9af8a2f4acd28c5dcb825d1ece2a0070b8/snuba/datasets/configuration/json_schema.py#L434-L439

However, in `dataset_builder`, it looks like it is required or else it raises KeyError: https://github.com/getsentry/snuba/blob/3d12cf9af8a2f4acd28c5dcb825d1ece2a0070b8/snuba/datasets/configuration/dataset_builder.py#L19

I've applied the fix to make it required which was the original intention I believe. 

To your question: Since the `Dataset` class requires `is_experimental`, I think we could make it "optional" by setting some default if not specified. However, I will differ to others with more context to decide how important it is to be explicit with this flag.

Reply:
My opinion is that explicit is better than implicit. A Dataset needs to know whether it is experimental or not, so we should explicitly codify that.